x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/final_readonly/fields/machine_learning_job_id/machine_learning_job_id.stories.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useQueryClient } from '@tanstack/react-query';
+import type { Story } from '@storybook/react';
+import { MachineLearningJobIdReadOnly } from './machine_learning_job_id';
+import type { DiffableAllFields } from '../../../../../../../../../common/api/detection_engine';
+import { FieldReadOnly } from '../../field_readonly';
+import { ThreeWayDiffStorybookProviders } from '../../storybook/three_way_diff_storybook_providers';
+import { GET_MODULES_QUERY_KEY } from '../../../../../../../../common/components/ml_popover/hooks/use_fetch_modules_query';
+import { GET_RECOGNIZER_QUERY_KEY } from '../../../../../../../../common/components/ml_popover/hooks/use_fetch_recognizer_query';
+import { GET_JOBS_SUMMARY_QUERY_KEY } from '../../../../../../../../common/components/ml/hooks/use_fetch_jobs_summary_query';
+
+export default {
+  component: MachineLearningJobIdReadOnly,
+  title:
+    'Rule Management/Prebuilt Rules/Upgrade Flyout/ThreeWayDiff/FieldReadOnly/machine_learning_job_id',
+};
+
+const mockedModulesData = [
+  {
+    id: 'security_auth',
+    jobs: [
+      {
+        id: 'auth_high_count_logon_events',
+        config: {
+          groups: [],
+          custom_settings: {
+            security_app_display_name: 'Spike in Logon Events',
+          },
+        },
+      },
+    ],
+  },
+];
+
+const mockedCompatibleModules = [
+  {
+    id: 'security_auth',
+  },
+];
+
+function MockMlData({ children }: { children: React.ReactNode }) {
+  const queryClient = useQueryClient();
+
+  queryClient.setQueryData([GET_JOBS_SUMMARY_QUERY_KEY, {}], []);
+
+  queryClient.setQueryData([GET_MODULES_QUERY_KEY, {}], mockedModulesData);
+
+  queryClient.setQueryData(
+    [GET_RECOGNIZER_QUERY_KEY, { indexPatternName: undefined }],
```

Comment:
nit: `indexPatternName: undefined` could be omitted since `undefined` is a default value for unset fields.

Reply:
https://github.com/elastic/kibana/pull/191499/commits/b56e497fd14cd126c17c0451cd778bff5418ccde